components/zcash_note_encryption/src/lib.rs

```diff
@@ -276,6 +295,10 @@ pub trait Domain {
     /// Returns `None` if `out_plaintext` does not contain a valid byte encoding of an
     /// `EphemeralSecretKey`.
     fn extract_esk(out_plaintext: &OutPlaintextBytes) -> Option<Self::EphemeralSecretKey>;
+
+    fn separate_tag_from_ciphertext(enc_ciphertext: Self::EncNoteCiphertextBytes) -> (Self::NotePlaintextBytes, [u8; AEAD_TAG_SIZE]);
```

Comment:
Attempted to define a trait for `EncNoteCiphertextBytes` so that this can be done via that trait instead of here:
`pub trait EncNoteCiphertextTrait<D: Domain> {
    fn separate(&self) -> (D::NotePlaintextBytes, [u8; AEAD_TAG_SIZE]);
}`

But then adding this trait in line 134 will require `EncNoteCiphertextTrait<Self>`, which doesn't work since the compiler does not know the size of Self at that point?